### PR TITLE
[stm32] [dma] [irq] Always inline IRQ handler to reduce IRQ latency

### LINKED ISSUE #963: [stm32] Interrupt handler indirection causes unnecessary latency
`MODM_ISR(DMA1_Channel{{N}})` calls nothing but `Dma1::Channel<DmaBase::Channel::Channel{{N}}>::interruptHandler();` in [`dma.cpp.in`](https://github.com/modm-io/modm/blob/develop/src/modm/platform/dma/stm32/dma.cpp.in#L18)

Unfortunately, these calls are not inlined. Adding an `inline` did not change anything for me. These calls are real calls, increasing interrupt latency unnecessarily.

[InterruptHandler](https://github.com/modm-io/modm/blob/develop/src/modm/platform/dma/stm32/dma.hpp.in#L311) is a jinja-generated C++ template ...

Why not jinja-generate this code directly into `dma.cpp` without the C++ template?

(Yes, I have to shave off some nanoseconds IRQ latency and I try to stay with modm for convenience.)

### PR DESCRIPTION
**Pull request by strongly-typed**
As mentioned by @chris-durand in #963

Shaves off 80ns on (slightly overclocked) STM32F107 ...

### CODE REVIEW

**Comment by salkinium** [comment]
> Shaves off 80ns on (slightly overclocked) STM32F107 ...

How much can you overclock it before it becomes unstable? Why not use something faster, like the F7/H7? Parts shortage?

**Comment by strongly-typed** [comment]
I have a (Japanese) encoder that needs a 18-bit serial protocol with 2.5 Msps over half-duplex RS485. Frames with 18 bits (1 start bit, 16 data bits, 1 stop bit, 1 to 4 frames) do not fit into U(SART), SPI, et cetera. That is why I use GPIO DMA triggered by a timer. I run a blue pill at 80 MHz to achieve 2.5 Msps (80/2.5 = 32).

I already have multiple STM32G474 on my desk and will do some custom hardware asap, but I wanted to test software and constraints before building the board. I need to quickly change from TX to RX. I use the `setTransferCompleteIrqHandler` and saw some cycles wasted there. Timings are good (enough) with the `[[gnu::always_inline]]`.

GPIO toggle works quite fine. I successfully read the encoder and can fully interpret the data. Data is protected by a CRC8 and  everything works on the STM32F103 (clone?) so far.

**Review by salkinium — APPROVED** [review]

File: src/modm/platform/dma/stm32/dma.hpp.in [hunk 1]
```diff
@@ -308,7 +308,7 @@ public:
 		 * of error the DMA channel will be disabled.
 		 */
 		static void
-		interruptHandler()
+		interruptHandler [[gnu::always_inline]] ()
```
Comment (salkinium) [inline src/modm/platform/dma/stm32/dma.hpp.in]:
```suggestion
		static void [[gnu::always_inline]]
		interruptHandler()
```
never seen your way before 🫥

Comment (rleh) [reply src/modm/platform/dma/stm32/dma.hpp.in]:
```suggestion
		modm_always_inline
		interruptHandler ()
```

Comment (strongly-typed) [reply src/modm/platform/dma/stm32/dma.hpp.in]: ... but it worked. Just pushed your suggestion

File: src/modm/platform/dma/stm32/dma.hpp.in [hunk 2]
```diff
@@ -307,8 +307,8 @@ public:
 		 * Reads the IRQ status and checks for error or transfer complete. In case
 		 * of error the DMA channel will be disabled.
 		 */
-		static void
-		interruptHandler()
+		modm_always_inline static void
+		interruptHandler
```
Comment (salkinium) [inline src/modm/platform/dma/stm32/dma.hpp.in]:
```suggestion
		interruptHandler()
```

Comment (strongly-typed) [reply src/modm/platform/dma/stm32/dma.hpp.in]: ... uuups ... fixed

**Review by chris-durand — APPROVED** [review]

**Review by rleh — APPROVED** [review]

**Pull request merged**